internal/gke/create.go

```diff
@@ -0,0 +1,304 @@
+package gke
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	gkev1 "github.com/rancher/gke-operator/pkg/apis/gke.cattle.io/v1"
+	gkeapi "google.golang.org/api/container/v1"
+)
+
+// Errors
+const (
+	cannotBeNilError            = "field [%s] cannot be nil for non-import cluster [%s]"
+	cannotBeNilForNodePoolError = "field [%s] cannot be nil for nodepool [%s] in non-nil cluster [%s]"
+)
+
+// Create creates an upstream GKE cluster.
+func Create(ctx context.Context, client *gkeapi.Service, config *gkev1.GKEClusterConfig) error {
+	err := validateCreateRequest(ctx, client, config)
+	if err != nil {
+		return err
+	}
+
+	createClusterRequest := newClusterCreateRequest(config)
+
+	_, err = client.Projects.
+		Locations.
+		Clusters.
+		Create(
+			LocationRRN(config.Spec.ProjectID, config.Spec.Region),
+			createClusterRequest).
+		Context(ctx).
+		Do()
+
+	return err
+}
+
+// CreateNodePool creates an upstream node pool with the given cluster as a parent.
+func CreateNodePool(ctx context.Context, client *gkeapi.Service, config *gkev1.GKEClusterConfig, nodePoolConfig *gkev1.NodePoolConfig) (Status, error) {
+	err := validateNodePoolCreateRequest(nodePoolConfig)
+	if err != nil {
+		return NotChanged, err
+	}
+
+	createNodePoolRequest, err := newNodePoolCreateRequest(
+		ClusterRRN(config.Spec.ProjectID, config.Spec.Region, config.Spec.ClusterName),
+		nodePoolConfig,
+	)
+	if err != nil {
+		return NotChanged, err
+	}
+
+	_, err = client.Projects.
+		Locations.
+		Clusters.
+		NodePools.
+		Create(
+			ClusterRRN(config.Spec.ProjectID, config.Spec.Region, config.Spec.ClusterName),
+			createNodePoolRequest,
+		).Context(ctx).Do()
+	if err != nil && strings.Contains(err.Error(), errWait) {
+		return Retry, nil
+	}
+	if err != nil {
+		return NotChanged, err
+	}
+
+	return Changed, nil
+}
+
+// newClusterCreateRequest creates a CreateClusterRequest that can be submitted to GKE
+func newClusterCreateRequest(config *gkev1.GKEClusterConfig) *gkeapi.CreateClusterRequest {
+
+	enableAlphaFeatures := config.Spec.EnableAlphaFeature != nil && *config.Spec.EnableAlphaFeature
+	request := &gkeapi.CreateClusterRequest{
+		Cluster: &gkeapi.Cluster{
+			Name:                  config.Spec.ClusterName,
+			Description:           config.Spec.Description,
+			InitialClusterVersion: *config.Spec.KubernetesVersion,
+			EnableKubernetesAlpha: enableAlphaFeatures,
```

Comment:
Whats `EnableKubernetesAlpha` ? I only see `EnableAlphaFeature` on spec.

Reply:
`EnableKubernetesAlpha` is what it's called in the Go SDK https://pkg.go.dev/google.golang.org/api/container/v1#Cluster

I can rename it in spec to be consistent with the Go SDK in the next PR.